src/core/stream_recv.c

```diff
@@ -873,14 +873,10 @@ QuicStreamRecvFlush(
     while (FlushRecv) {
         CXPLAT_DBG_ASSERT(!Stream->Flags.SentStopSending);
 
-        CxPlatRwLockAcquireExclusive(&Stream->ReceiveCompleteInlineLock);
-        Stream->Flags.ReceiveCallActive = TRUE;
         //
-        // Save the current value of RecvCompletionLength to check if it has
-        // changed during the app callback.
+        // Set the top bit of RecvCompletionLength to indicate that there is an active receive.
         //
-        uint64_t RecvCompletionLength = Stream->RecvCompletionLength;
-        CxPlatRwLockReleaseExclusive(&Stream->ReceiveCompleteInlineLock);
+        InterlockedExchangeAdd64((int64_t*)&Stream->RecvCompletionLength, 0x8000000000000000);
```

Comment:
```suggestion
        BOOLEAN RecvCompletionPending =
            InterlockedExchangeAdd64((int64_t*)&Stream->RecvCompletionLength, 0x8000000000000000) != 0;
```

Reply:
Then, down below only go into the `if` that calls `QuicStreamReceiveComplete` if this is `FALSE`.